### PR TITLE
feat: add dynamic favicon

### LINKED ISSUE #18: Add dynamic favicon

### LINKED ISSUE #23: `<ColorPicker>` needs a test
[This functionality](https://github.com/fmaclen/chromabox/blob/main/src/routes/ColorCard.svelte#L16-L18) is untested:

```ts
$effect(() => {
  color = colord(hex);
});
```
This `$effect` makes the `<ColorPicker>` reactive. Commenting out this function doesn't break any tests.

Need a test that will click somewhere inside the color picker and assert all the other fields have been updated accordingly.

### PR DESCRIPTION
**Pull request by fmaclen**
- favicon changes as colors get added to the palette
- Closes #23 
- Closes #18

### CODE REVIEW

File: e2e/favicon.test.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			await picker.click({
+				position: {
+					x: box.width / 2,
+					y: box.height / 2
+				}
+			});
+		}
+
+		// Check the value in HEX has changed from the color picker
+		expect(await colorCard.getByLabel('HEX', { exact: true }).inputValue()).not.toBe(firstColorHex);
```
Comment (fmaclen) [inline e2e/favicon.test.ts:54]: This satisfies #23

File: src/routes/ColorCard.svelte [hunk 2]
```diff
@@ -2,7 +2,7 @@
 	import { colord, type Colord } from 'colord';
 	import ColorPicker, { ChromeVariant } from 'svelte-awesome-color-picker';
 
-	let { color }: { color: Colord } = $props();
+	let { color = $bindable() }: { color: Colord } = $props();
```
Comment (fmaclen) [inline src/routes/ColorCard.svelte:5]: Had to make it `$bindlable()` so I could get reactivity in the parent `+page.svelte` and maintain `Favicon` updated whenever an existing color changes.

File: src/routes/Favicon.svelte [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		const svg = `
+    <svg width="475" height="598" viewBox="0 0 475 598" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <path class="left" d="M233 0L0 161.5L235.5 327L474.5 161.5L233 0Z" fill="${box.top}"/>
+    <path class="top" d="M234.5 327L0 161.5V430L234.5 597.5V327Z" fill="${box.left}"/>
+    <path class="right" d="M474.5 162L235 327.5V596L472.5 424L474.5 162Z" fill="${box.right}"/>
+    </svg>
+    `;
+
+		faviconElement?.setAttribute('href', `data:image/svg+xml,${encodeURIComponent(svg)}`);
+	}
```
Comment (fmaclen) [inline src/routes/Favicon.svelte:36]: This URI encodes the SVG in the `href` attribute of the metatag.

**Review by GregoMac1 — APPROVED** [review]
This is a really nice detail to have in the app. I hadn't thought about it and now that I see it I think it's great!

**Pull request merged**